modules/tensor_mechanics/src/actions/TensorMechanicsActionBase.C

```diff
@@ -57,6 +57,7 @@ validParams<TensorMechanicsActionBase>()
   params.addParam<bool>("add_variables", false, "Add the displacement variables");
   params.addParam<std::vector<MaterialPropertyName>>(
       "eigenstrain_names", "List of eigenstrains to be applied in this strain calculation");
+  params.addParam<bool>("use_ad", false, "Flag to use AD objects when possible");
```

Comment:
This is cool!  I would suggest spelling this out though, because not everyone is going to know what 'AD' is. I would prefer that the parameter be `use_automatic_differentiation`, but at a minimum, it should be spelled out in the comment.

Reply:
Good point